libs/velo-external-db-core/src/service/schema.spec.ts

```diff
@@ -1,90 +1,285 @@
+            expect(compareColumnsInDbAndRequest(columnsInDb, [...columnsInRequest, newColumn]).columnsToRemove).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, []).columnsToRemove).toEqual(columnsInDb.map(f => f.field))
+            expect(compareColumnsInDbAndRequest(columnsInDb, [newColumn]).columnsToRemove).toEqual(columnsInDb.map(f => f.field))
+        })
 
-        await expect(env.schemaService.create(ctx.collectionName)).resolves.toEqual({})
-    })
+        // TODO: move it to schema utils tets
+        test('compareColumnsInDbAndRequest function - change column type', async() => {
+            const columnsInDb = [{
+                field: ctx.column.name,
+                type: 'text'
+            }]
 
-    test('add column for collection name', async() => {
-        driver.givenAllSchemaOperations()
-        driver.expectCreateColumnOf(ctx.column, ctx.collectionName)
-        schema.expectSchemaRefresh()
+            const columnsInRequest = [{
+                key: ctx.column.name,
+                type: fieldTypeToWixDataEnum('text'),
+            }]
 
-        await expect(env.schemaService.addColumn(ctx.collectionName, ctx.column)).resolves.toEqual({})
-    })
+            const changedColumnType = {
+                key: ctx.column.name,
+                type: fieldTypeToWixDataEnum('number')
+            }
 
-    test('remove column from collection name', async() => {
-        driver.givenAllSchemaOperations()
-        driver.expectRemoveColumnOf(ctx.column, ctx.collectionName)
-        schema.expectSchemaRefresh()
+            expect(compareColumnsInDbAndRequest([], []).columnsToChangeType).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, columnsInRequest).columnsToChangeType).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, [changedColumnType]).columnsToChangeType).toEqual(WixFormatFieldsToInputFields([changedColumnType]))
+        })
 
-        await expect(env.schemaService.removeColumn(ctx.collectionName, ctx.column.name)).resolves.toEqual({})
-    })
+        test('update collection - add new columns', async() => { 
+            const newFields = [{
+                key: ctx.column.name,
+                type: fieldTypeToWixDataEnum(ctx.column.type),
+            }]
 
-    test('collections without _id column will have read-only capabilities', async() => {
-        driver.givenAllSchemaOperations()
-        driver.givenListResult(ctx.dbsWithoutIdColumn)
+            driver.givenAllSchemaOperations()
+            schema.expectSchemaRefresh()            
+            driver.givenFindResults([ { id: ctx.collectionName, fields: [] } ])
 
-        await expect( env.schemaService.list() ).resolves.toEqual( schemasWithReadOnlyCapabilitiesFor(ctx.dbsWithoutIdColumn) )
-    })
+            await env.schemaService.update({ id: ctx.collectionName, fields: newFields })
+
+
+            expect(driver.schemaProvider.addColumn).toBeCalledTimes(1)    
```

Comment:
I don't know if it's necessary, but if you think you do, you might test it in the other tests as well.

Reply:
I did this to make sure that addColumn is called once and not more.